### DISCUSSION TITLE
The settings should have white list (where FL can limit its search) and black list (banned folders)

### DISCUSSION THREAD

**Discussion by raindropsfromsky** [Flow-Launcher/Flow.Launcher#1175] [Ideas]
The settings should have a provision where the user can define a **white list** (FL must limit its search within this list) and a **black list** (banned folders where FL should not search at all).

This will show more relevant results to the user and eliminate the clutter; making it easier to use FL.

This is not for the sake of speed (FL is blazing fast even when all folders are scanned), but for the sake of decluttered output.

**Comment by taooceros** [comment]
If you are using the explorer plugin, you can open the context menu of a specific folder, which includes an option to exclude that from indexing search. Or you can go to index option to exclude specific folder.
For everything plugin, you can go to everything directly to exclude paths.

White list can also be done for everything plugin, but probably not for the windows index.

**Reply by raindropsfromsky** [reply]
I have not quite understood the nuances here:

I am unable to try the **Explorer** plugin, because my AVG Antivirus moved FL to quarantine, and reported that it is infected with IDP.generic. (I will be raising a bug about this separately).

The **Everything** app appears as a separate app altogether:
1. It has its own launch icon in the System Tray. 
2. When I click this icon, Everything GUI appears, which does not look integrated with FL in any way.

I expected the integration with FL in the sense that I enter the search string in FL's search bar and use a specific action keyword to pass it to **Everything**, and then **Everything** passes the results back to FL's search results.  At least, if this happens in the background, I cannot make it out by looking at the results.

**Reply by taooceros** [reply]
You need to install the everything plugin to use the integration with everything now. I have no idea about the infected situation. If you delete the Explorer plugin, will the antivirus software view flow as safe?

**Reply by raindropsfromsky** [reply]
Yes, I did activate the **Everything** plugin, but the only effect I could see was a separately launchable app (independent GUI and its own icon in SysTray).

I am stuck with this antivirus quarantine issue, so cannot report further.

Hopefully, I will be able to continue once this is sorted out.
Kindly bear with me till then, and keep this issue in suspension!
Thanks in advance!

**Reply by taooceros** [reply]
Well the everything plugin is allowing you to search the whole system in Flow's search bar (it use everything as backend though). So everything is a separate app, but it also provide service for searching. 
I don't know why the antivirus reports the issue. Probably because we have some hook that is used to interrupt some hot key? Anyway feel free to read the code and the publish process. As far as I know it should be a false positive.

**Reply by raindropsfromsky** [reply]
An AVG support agent has responded to my false positive report and taken up the issue.
I have sent the AVG screenshot to him.

**Reply by taooceros** [reply]
Thank you!
